cli/pcluster/config/validators.py

```diff
@@ -839,3 +839,41 @@ def intel_hpc_validator(param_key, param_value, pcluster_config):
         )
 
     return errors, warnings
+
+
+def custom_chef_cookbook_validator(param_key, param_value, pcluster_config):
+    """
+    Validate custom_chef_cookbook parameter.
+
+    Assumes that the S3 bucket where the cookbook is stored should be listed in
+    s3_read_resource explicitly, or s3_read_resource should contain a wildcard.
+    """
+    errors = []
+    warnings = []
+
+    if urlparse(param_value).scheme in ["https", "http"]:
+        errors, warnings = url_validator(param_key, param_value, pcluster_config)
+
+    elif urlparse(param_value).scheme == "s3":
+
+        # Check that bucket is in s3_read_resource.
+        cluster_section = pcluster_config.get_section("cluster")
+        s3_read_resource = cluster_section.get_param_value("s3_read_resource")
```

Comment:
I'd check also the `s3_read_write_resource`.

Reply:
Done.